zh-CN: update translation Glossary/IIFE (#20603)
mdn/translated-content: files/zh-cn/glossary/iife/index.md

### 执行一个异步函数

> [`async`](/zh-CN/docs/Web/JavaScript/Reference/Operators/async_function) 立即调用函数表达式允许你在比较旧的浏览器或者 JavaScript 运行环境没有[顶层 await](/zh-CN/docs/Web/JavaScript/Reference/Operators/await#top_level_await) 中使用 [`await`](/zh-CN/docs/Web/JavaScript/Reference/Operators/async_function) 和 [`for-await`](/zh-CN/docs/Web/JavaScript/Reference/Statements/for-await...of)：

Reviewer: ```suggestion
[`async`](/zh-CN/docs/Web/JavaScript/Reference/Operators/async_function) 立即调用函数表达式允许你在比较旧的浏览器中或者 JavaScript 运行环境没有[顶层 await](/zh-CN/docs/Web/JavaScript/Reference/Operators/await#在顶层使用_await) 时使用 [`await`](/zh-CN/docs/Web/JavaScript/Reference/Operators/await) 和 [`for-await`](/zh-CN/docs/Web/JavaScript/Reference/Statements/for-await...of)：
```

### 模块模式

我们也可以使用立即调用函数表达式来创建私有和公有变量、方法。对于模块模式的更复杂用法和立即调用函数表达式的其他用途，你可以参考 Addy Osmani 的《Learning JavaScript Design Patterns》这本书。

```js
const makeWithdraw = (balance) => {
  ((copyBalance) => {
    let balance = copyBalance; // 这个变量是私有的
    const doBadThings = () => {
      console.log("我要用你的钱做坏事");
    };
    doBadThings();
    return {
      withdraw(amount) {
        if (balance >= amount) {
          balance -= amount;
          return balance;
        }
        return "不够钱";
      },
    };
  })(balance);

  const firstAccount = makeWithdraw(100); // "我要用你的钱做坏事"
  console.log(firstAccount.balance); // undefined
  console.log(firstAccount.withdraw(20)); // 80
  console.log(firstAccount.withdraw(30)); // 50
  console.log(firstAccount.doBadThings); // undefined；这个方法是私有的
  const secondAccount = makeWithdraw(20); // "我要用你的钱做坏事"
  console.log(secondAccount.withdraw(30)); // "不够钱"
  console.log(secondAccount.withdraw(20)); // 0
}
```

### ES6 之前在 For 循环中使用 var

> 在 **ES6** 引入 **let** 和 **const** 声明和块级作用域之前，我们可以在一些旧代码中看到立即调用函数表达式的以下用法。通过 **var** 声明变量，只有函数作用域和全局作用域。假设我们创建两个按钮，文本按钮 0 和按钮 1，并且当我们点击它们时，想让它们弹窗 0 和 1。下面的代码不能起作用：

Reviewer: 下面代码的 `button.innerText`（106、120、136 行）可以做适当修改

```suggestion
在 **ES6** 引入 **let** 和 **const** 声明和块级作用域之前，我们可以在一些旧代码中看到立即调用函数表达式的以下用法。通过 **var** 声明变量，只有函数作用域和全局作用域。假设我们创建两个按钮，分别称为“按钮 0”和“按钮 1”，并且当我们点击它们时，想让它们分别弹窗提示 0 和 1。下面的代码不能起作用：
```

console.log(i); // 2
```

> 当点击时，按钮 0 和按钮 1 都会弹窗 2，因为 `i` 是全局的，并且值为 2。在 ES6 之前为了解决这个问题，我们可以使用立即调用函数表达式模式：

Reviewer: ```suggestion
当点击时，按钮 0 和按钮 1 都会弹窗提示“2”，因为 `i` 是全局的，且最终的值为 2。在 ES6 之前，为了解决这个问题，我们可以使用立即调用函数表达式模式：
```

console.log(i); // 2
```

> 当点击时，按钮 0 和按钮 1 会弹窗 0 和 1，变量 `i` 是全局的。更简单的是使用 **let** 声明变量：

Reviewer: ```suggestion
当点击时，按钮 0 和按钮 1 会分别弹窗提示 0 和 1，变量 `i` 是全局的。更简单的是使用 **let** 声明变量：
```

console.log(i); // Uncaught ReferenceError: i is not defined.
```

> 当我们点击按钮时，就会弹窗 0 和 1。

Reviewer: ```suggestion
当我们点击按钮时，这些按钮会分别弹窗提示 0 和 1。
```
